dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/LambdaEventDAO.java

```diff
@@ -39,6 +51,44 @@ public void delete(LambdaEvent lambdaEvent) {
         session.flush();
     }
 
+    private List<Predicate> processQuery(String filter, String sortCol, String sortOrder, CriteriaBuilder cb, CriteriaQuery query, Root<LambdaEvent> event) {
+        List<Predicate> predicates = new ArrayList<>();
+        if (!Strings.isNullOrEmpty(filter)) {
+            predicates.add(
+                    // ensure we deal with null values and then do like queries on those non-null values
+                    cb.or(cb.and(cb.isNotNull(event.get("message")), cb.like(cb.upper(event.get("message")), "%" + filter.toUpperCase() + "%")), //
+                            cb.and(cb.isNotNull(event.get("githubUsername")), cb.like(cb.upper(event.get("githubUsername")), "%" + filter.toUpperCase() + "%")), //
+                            cb.and(cb.isNotNull(event.get("repository")), cb.like(cb.upper(event.get("repository")), "%" + filter.toUpperCase() + "%")), //
+                            cb.and(cb.isNotNull(event.get("type")), cb.like(cb.upper(event.get("type")), "%" + filter.toUpperCase() + "%")), //
+                            cb.and(cb.isNotNull(event.get("reference")), cb.like(cb.upper(event.get("reference")), "%" + filter.toUpperCase() + "%")), //
+                            cb.and(cb.isNotNull(event.get("deliveryId")), cb.like(cb.upper(event.get("deliveryId")), "%" + filter.toUpperCase() + "%"))));
+        }
+
+        if (!Strings.isNullOrEmpty(sortCol)) {
```

Comment:
Optional, since I don't think we always do this, it could be nice to have a default sort order if one is not specified.

Reply:
I added it as a `@DefaultValue` onto the endpoints. Should I also add it in this function as well? https://github.com/dockstore/dockstore/blob/b2596e8ee19afafd1160ae308ee1c9aa92dcb251/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/LambdaEventResource.java#L71

Reply:
I don't the `@DefaultValue` annotation would do anything here -- the annotation only works for the endpoints Jersey endpoints.

The value could still be empty though, although it would be a bit of an edge case -- the user would have to specify an empty value, or somebody accidentally set it to null, so my proposal is to have an `else` just in case it is null or empty.